cmd/kubeadm/app/phases/etcd/local.go

```diff
@@ -176,7 +179,11 @@ func GetEtcdPodSpec(cfg *kubeadmapi.ClusterConfiguration, endpoint *kubeadmapi.A
 		etcdVolumeName:  staticpodutil.NewVolume(etcdVolumeName, cfg.Etcd.Local.DataDir, &pathType),
 		certsVolumeName: staticpodutil.NewVolume(certsVolumeName, cfg.CertificatesDir+"/etcd", &pathType),
 	}
+	// TODO: probeHostname is obtained from listen-metrics-urls but the argument is created later with getEtcdCommand
```

Comment:
I was thinking in different ways of dealing with this chicken an egg problem but didn't find a better one. Will nice some guidance here, just choose the simplest solution

Reply:
GetEtcdProbeEndpoint() can be made to have a bool flag for ipv6:

```
probeHostname, probePort, probeScheme := staticpodutil.GetEtcdProbeEndpoint(&cfg.Etcd, utilsnet.IsIPv6String(endpoint.AdvertiseAddress))

GetEtcdProbeEndpoint(cfg *kubeadmapi.Etcd, bool isIPv6) (string, int, v1.URIScheme)
```

this is better than the override after the function call, but requires minor unit test changes.

Reply:
done